### PR TITLE
Mqtt resolve-redux

### PR DESCRIPTION
**Pull request by MrCheater**
Implements features:
- Connect to multiple aggregate ids in view models connector
- Lifecycle actions for read/view-model connectors (Low level API)
- Subscribe adapters support (MQTT, socket.io)
- Fault tolerance for network errors

Fixes issues:
https://github.com/reimagined/resolve/issues/594
https://github.com/reimagined/resolve/issues/596
https://github.com/reimagined/resolve/issues/670
https://github.com/reimagined/resolve/issues/664

Breaking changes:
- Query support now GET and POST queries in new manner: `/api/query/viewModelName/aggregateIds` for view models and `/api/query/readModelName/resolverName` for read models with optional arguments in request body (for POST) or query string (for GET) requests
- Resolve-query provides result for view-model in manner `{ state, aggregateVersionsMap }` , which allows to filter obsolete or duplicate events on unstable network on client browser

### CODE REVIEW

File: examples/hacker-news/client/containers/PagePreloader.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      match
+    } of matchRoutes(routes, route)) {
+      if (match.path !== '/') {
+        children.push(<Component key={index++} match={match} />)
+      }
+    }
+  }
+  return (
+    <div style={{ display: 'none' }}>
+      {children.length > 0 ? children : null}
```
Comment (TatyanaRyzh) [inline examples/hacker-news/client/containers/PagePreloader.js]: {children.length > 0 ? children : null}
=>
{children.length ? children : null}

File: examples/hacker-news/client/containers/PagePreloader.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { matchRoutes } from 'react-router-config'
+
+import routes from '../routes'
+
+const PagePreloader = ({ route }) => {
+  let children = []
+  if (route != null) {
```
Comment (TatyanaRyzh) [inline examples/hacker-news/client/containers/PagePreloader.js]: if (route != null) {
=>
if (route) {

File: examples/hacker-news/client/containers/LoginInfo.js [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
-import { NavLink } from 'react-router-dom'
+import { Link as NormalLink } from 'react-router-dom'
 
 import Splitter from '../components/Splitter'
-import { connect } from 'react-redux'
 import * as userActions from '../actions/userActions'
-import { bindActionCreators } from 'redux'
 
-const Link = styled(NavLink)`
+const Link = styled(NormalLink)`
```
Comment (TatyanaRyzh) [inline examples/hacker-news/client/containers/LoginInfo.js]: I guess it'll be better to have NormalLink as Link, and Link as '%Smth%Link' (for example. CustomLink or something like this)

File: examples/hacker-news/client/middlewares/index.js [hunk 4]
```diff
@@ -1,40 +1,74 @@
...
 const storyCreateMiddleware = () => next => action => {
-  if (action.type === SEND_COMMAND) {
-    if (action.command.type === 'createStory') {
-      if (action.command.ok) {
-        window.location = `${rootDirectory}/storyDetails/${action.aggregateId}`
-      } else if (action.command.error) {
-        window.location = `${rootDirectory}/error?text=Failed to create a story`
-      }
-    }
+  if (
```
Comment (TatyanaRyzh) [inline examples/hacker-news/client/middlewares/index.js]: =>
if (action.commandType === 'createStory') {
window.location = action.type === SEND_COMMAND_SUCCESS ? `${rootDirectory}/storyDetails/${action.aggregateId}` : `${rootDirectory}/error?text=Failed to create a story`
}

File: examples/hacker-news/client/middlewares/index.js [hunk 5]
```diff
@@ -1,40 +1,74 @@
...
-        type: OPTIMISTIC_STORY_UPVOTED,
-        storyId: action.aggregateId
-      })
-    }
-    if (action.command.type === 'unvoteStory') {
-      store.dispatch({
-        type: OPTIMISTIC_STORY_UNVOTED,
-        storyId: action.aggregateId
-      })
+  if (
```
Comment (TatyanaRyzh) [inline examples/hacker-news/client/middlewares/index.js]: SEND_COMMAND_SUCCESS to one 'if'

File: examples/hacker-news/client/reducers/prefetchRoute.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { ROUTE_CHANGED } from '../actions/actionTypes'
+
+const prefetchRoute = (state = null, action) => {
+  switch (action.type) {
```
Comment (TatyanaRyzh) [inline examples/hacker-news/client/reducers/prefetchRoute.js]: Switch for two options?

File: examples/hacker-news/test/functional/2.story.js [hunk 7]
```diff
@@ -36,8 +36,13 @@ test('add comment', async (t /*: TestController */) => {
 
   await t.click(titleLink)
 
-  await t.typeText('textarea', 'first comment')
-  await t.click('button')
+  await t.wait(5000) // TODO Fix reactivity
```
Comment (TatyanaRyzh) [inline examples/hacker-news/test/functional/2.story.js]: What do you mean about fixes for reactivity?

**Comment by Himura2la** [comment]
To be Checked:

- [ ] packages/resolve-redux/README.md

- [ ] packages/subscribe-adapters/resolve-subscribe-mqtt/README.md

- [ ] packages/subscribe-adapters/resolve-subscribe-socket.io/README.md

**Comment by Himura2la** [comment]
@MrCheater, please allow pushing

File: examples/hacker-news/client/containers/Submit.js [hunk 8]
```diff
@@ -1,11 +1,11 @@
 import React from 'react'
 import uuid from 'uuid'
 import { connect } from 'react-redux'
+import { connectResolveAdvanced } from 'resolve-redux'
```
Comment (ralkov) [inline examples/hacker-news/client/containers/Submit.js]: change connectResolveAdvanced

**Comment by Himura2la** [comment]
I'm not able to push commits here. so I created a PR into this PR: https://github.com/MrCheater/resolve/pull/1

**Pull request merged**